src/world/world.rs

```diff
@@ -146,23 +146,10 @@ impl<N: Scalar> World<N> {
     pub fn add_body(&mut self, rb: RigidBody<N>) -> RigidBodyHandle<N> {
         let position = rb.position().clone();
         let shape = rb.shape().clone();
+        let groups = (*(*rb.collision_groups())).clone();
```

Comment:
I agree with you, that's ugly… keeping the `.as_collision_groups()` method is probably better than the `Deref` trait after all ^^